sirepo/sim_data/ml.py

```diff
@@ -62,3 +63,57 @@ def _lib_file_basenames(cls, data):
         if name:
             return [cls.lib_file_name_with_model_field("dataFile", "file", name)]
         return []
+
+
+class ArchiveManager:
+    import contextlib
+
+    _ARCHIVE_EXTENSIONS = (
+        ".tar.gz",
+        ".zip",
+    )
+
+    def __init__(self, file_path):
+        import zipfile
+        import tarfile
+
+        self.file_path = file_path
+        self.filename = file_path if isinstance(file_path, str) else file_path.basename
+        if self._is_archive_type(".zip"):
+            self.dir_check = "is_dir"
+            self.extractor = "open"
+            self.file_ctx = zipfile.ZipFile
+            self.item_name = "filename"
+            self.lister = "infolist"
+        if self._is_archive_type(".tar.gz"):
+            self.dir_check = "isdir"
+            self.extractor = "extractfile"
+            self.file_ctx = tarfile.open
+            self.item_name = "name"
+            self.lister = "getmembers"
+
+    def _is_archive_type(self, ext):
+        return self.filename.endswith(ext)
+
+    def is_archive(self):
+        return any(
+            [self._is_archive_type(s) for s in ArchiveManager._ARCHIVE_EXTENSIONS]
+        )
+
+    @contextlib.contextmanager
+    def data_ctx(self, data_path):
+        if not self.is_archive():
+            yield open(self.file_path)
+        else:
+            with self.file_ctx(self.file_path, mode="r") as f:
```

Comment:
I don't like the name data_ctx. It doesn't tell me anything. It's return an open file that can be read. 

However, I see how it is used, and I think what you really want is something that matches your problem better, eg. return a numpy array or csv depending on what the data is.

Reply:
[5895f25](https://github.com/radiasoft/sirepo/pull/4736/commits/5895f2509fbb49bd496a21aaa025dbd3c8768977)